hkmc2/shared/src/test/mlscript/codegen/Quasiquotes.mls

```diff
@@ -64,2103 +64,114 @@
-//│ ═══[COMPILATION ERROR] No definition found in scope for member 'tmp'
-//│ FAILURE: Unexpected compilation error
-//│ FAILURE LOCATION: lookup_! (Scope.scala:114)
-//│ FAILURE INFO: Tuple2:
-//│   _1 = Tuple2:
-//│     _1 = $arr
-//│     _2 = class hkmc2.semantics.TempSymbol
-//│   _2 = Scope:
-//│     parent = N
-//│     curThis = S of S of globalThis:globalThis
-//│     bindings = HashMap($block$res -> block$res10, $block$res -> block$res15, $runtime -> runtime, $definitionMetadata -> definitionMetadata, $prettyPrint -> prettyPrint, $Term -> Term, $Block -> Block, $Shape -> Shape, $block$res -> block$res11, member:Predef -> Predef, $block$res -> block$res, term:Term -> Term1, $block$res -> block$res12, $block$res -> block$res1, $block$res -> block$res2, $block$res -> block$res3, $block$res -> block$res4, $block$res -> block$res5, $block$res -> block$res13, $block$res -> block$res6, $block$res -> block$res7, $tmp -> tmp1, $block$res -> block$res8, $block$res -> block$res14, $block$res -> block$res9, $tmp -> tmp, f -> f)
-//│ ═══[COMPILATION ERROR] No definition found in scope for member 'arr'
-//│ FAILURE: Unexpected compilation error
-//│ FAILURE LOCATION: lookup_! (Scope.scala:114)
-//│ FAILURE INFO: Tuple2:
-//│   _1 = Tuple2:
-//│     _1 = $tmp
-//│     _2 = class hkmc2.semantics.TempSymbol
-//│   _2 = Scope:
-//│     parent = N
-//│     curThis = S of S of globalThis:globalThis
-//│     bindings = HashMap($block$res -> block$res10, $block$res -> block$res15, $runtime -> runtime, $definitionMetadata -> definitionMetadata, $prettyPrint -> prettyPrint, $Term -> Term, $Block -> Block, $Shape -> Shape, $block$res -> block$res11, member:Predef -> Predef, $block$res -> block$res, term:Term -> Term1, $block$res -> block$res12, $block$res -> block$res1, $block$res -> block$res2, $block$res -> block$res3, $block$res -> block$res4, $block$res -> block$res5, $block$res -> block$res13, $block$res -> block$res6, $block$res -> block$res7, $tmp -> tmp1, $block$res -> block$res8, $block$res -> block$res14, $block$res -> block$res9, $tmp -> tmp, f -> f)
-//│ ═══[COMPILATION ERROR] No definition found in scope for member 'tmp'
 //│ = Lam(
-//│   [
-//│     Let(
-//│       Cons(Else(Ref(Symbol("x"))), Else(Ref(Symbol("x")))),
-//│       Cons(Else(Ref(Symbol("x"))), Else(Ref(Symbol("x")))),
-//│       Cons(Else(Ref(Symbol("x"))), Else(Ref(Symbol("x"))))
-//│     )
-//│   ],
+//│   [Symbol("x")],
 //│   IfLike(
 //│     If,
 //│     Let(
-//│       Cons(Else(Ref(Symbol("x"))), Else(Ref(Symbol("x")))),
-//│       Cons(Else(Ref(Symbol("x"))), Else(Ref(Symbol("x")))),
-//│       Cons(Else(Ref(Symbol("x"))), Else(Ref(Symbol("x"))))
+//│       Symbol("scrut"),
+//│       App(Builtin("==="), Tup([Ref(Symbol("x")), Lit(0)])),
+//│       Cons(
+//│         Branch(Ref(Symbol("scrut")), LitPattern(true), Else(Lit(1))),
+//│         Else(Ref(Symbol("x")))
+//│       )
 //│     )
 //│   )
 //│ )
 
-:fixme // FIXME: weird error
+
+:ge
+`1 `== `2
```

Comment:
Let's change these to `` `=== ``, then.

But it's strange that we can't use a global thing like `Predef` – that's not truly a cross-stage reference.

Reply:
It is because we need to compute which files should be included in the next stage. So, we need to use `import.meta` to get the relative path. In this example, `==` in `Predef` shadows the default one.  That's why we get the error.

Reply:
I still don't get it (see my other message).